### PR TITLE
Fix paging iterator for missing next_marker cases

### PR DESCRIPTION
**Pull request by mattwiller**
Some Box API endpoints do not return any `next_marker` field when the collection has finished paging; in this case, we should treat the collection as finished, buffer the data in the response, and let the user iterate over the buffer.

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @mattwiller has signed the CLA. Thanks for the pull request!

**Comment by mattwiller** [comment]
Fixes #177

File: lib/util/paging-iterator.js [hunk 1]
```diff
@@ -81,11 +83,13 @@ class PagingIterator {
 			this.nextField = PAGING_MODES.MARKER;
 			this.nextValue = data.next_marker;
 		} else {
-			throw new Error('Unable to determine paging strategy for response!');
+			// throw new Error('Unable to determine paging strategy for response!');
+			this.nextField = PAGING_MODES.MARKER;
+			this.nextValue = null;
+			this.done = true;
```
Comment (jmoldow) [inline lib/util/paging-iterator.js]: Why do you need this here?

Wouldn't it get set to `true` in the same way as if `data.next_marker` were `null` in the `else if` block?

**Comment by mattwiller** [comment]
@jmoldow You're right!  I removed the unnecessary `done` setting and added a comment to clarify what this case is doing.

**Pull request merged**